pros/src/sensors/imu.rs

```diff
@@ -0,0 +1,414 @@
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+use pros_sys::{PROS_ERR, PROS_ERR_F};
+use snafu::Snafu;
+
+use crate::error::{bail_on, map_errno, PortError};
+
+pub const IMU_RESET_TIMEOUT: Duration = Duration::from_secs(3);
+pub const IMU_MIN_DATA_RATE: Duration = Duration::from_millis(5);
+
+/// Represents a smart port configured as a V5 inertial sensor (IMU)
+#[derive(Debug, Clone, Copy)]
+pub struct InertialSensor {
+    port: u8,
+}
+
+impl InertialSensor {
+    /// Create a new inertial sensor from a smart port index.
+    pub fn new(port: u8) -> Result<Self, InertialError> {
+        let sensor = Self { port };
+        sensor.status()?;
+        Ok(sensor)
+    }
+
+    /// Calibrate IMU.
+    ///
+    /// This takes approximately 2 seconds, and is blocking until the IMU status flag is set properly.
+    /// There is additionally a 3 second timeout that will return [`InertialError::CalibrationTimedOut`] if the timeout is exceeded.
+    pub fn calibrate_blocking(&self) -> Result<(), InertialError> {
+        unsafe {
+            bail_on!(PROS_ERR, pros_sys::imu_reset_blocking(self.port));
+        }
+        Ok(())
+    }
+
+    /// Calibrate IMU asynchronously.
+    ///
+    /// There a 3 second timeout that will return [`InertialError::CalibrationTimedOut`] if the timeout is exceeded.
+    pub fn wait_until_calibrated(&self) -> InertialCalibrateFuture {
```

Comment:
This should be named `calibrate_async`. The motor function `wait_untill_stopped` is named that way because there is no blocking equivalent.